code/obj/submachine/robotics.dm

```diff
@@ -136,6 +136,35 @@
 			selection:contained = 0
 			elecflash(user)
 
+
+/obj/item/borg_lamp_manufacturer
+	name = "Miniaturized Lamp Manufacturer"
+	desc = "A small manufacturing unit for cyborgs to produce and (re)place lamps in existing fittings."
+	icon = 'icons/obj/items/device.dmi'
+	icon_state = "borglampman-tube"
+	var/prefix = "borglampman"
+
+	var/cost_broken = 50 //For broken/burned lamps (the old lamp gets recycled in the tool)
+	var/cost_empty = 75
+	var/setting = "tube" //Currently either tube or bulb
+	var/dispensing = /obj/item/light/tube
+
+	attack_self(var/mob/user as mob)
+		switch (src.setting) //This should be relatively easily expandable I think
```

Comment:
Why do we need to have a toggle between tube and bulb? Would it not make more sense to have the fitting define what type gets inserted when it's attacked by this? You can't insert bulbs into tube fittings or vice versa, so seems unnecessary to manage it.

Reply:
Fair enough. I've reworked it to cycle colours (the ones from the general fab) instead.

Reply:
Yay! Way more fun!